packages/mask/src/components/shared/SelectRecipients/SelectRecipientsDialog.tsx

```diff
@@ -130,13 +166,38 @@ export function SelectRecipientsDialogUI(props: SelectRecipientsDialogUIProps) {
         setSearchInput('')
         onSearch('')
     }
-    const handleSubmit = () => {
+
+    const handleSubmit = useCallback(async () => {
+        props.onSetSelected([...selectedAllProfiles])
+        for (const item of selectedAllProfiles) {
+            await attachNextIDToProfile(item as ProfileInformationFromNextID)
+        }
         props.onSubmit()
         setSearchInput('')
         onSearch('')
-    }
+    }, [selectedAllProfiles])
 
-    const selectedPubkeyList = compact(props.selected.map((x) => x.linkedPersona?.publicKeyAsHex))
+    const onSelectedProfiles = useCallback((item: Profile, checked: boolean) => {
+        if (checked) {
+            setSelectedAllProfiles((profiles) => [...profiles, item])
+        } else
+            setSelectedAllProfiles((profiles) =>
+                profiles.filter((x) => x.linkedPersona?.publicKeyAsHex !== item.linkedPersona?.publicKeyAsHex),
+            )
```

Comment:
If none of the profiles gets modified (you should double check)
```suggestion
            setSelectedAllProfiles((profiles) =>
                profiles.filter((x) => x !== item),
            )
```